src/patternfly/components/Table/table-grid.scss

```diff
@@ -437,11 +437,11 @@
 
   // Toggles
   .pf-c-table__toggle {
-    .pf-c-button > * {
+    .pf-c-button .pf-c-table__toggle-icon {
       transition: var(--pf-c-table__toggle__icon--Transition);
     }
 
-    .pf-c-button.pf-m-expanded > * {
+    .pf-c-button.pf-m-expanded .pf-c-table__toggle-icon {
       transform: var(--pf-c-table__toggle--m-expanded__icon--Transform);
     }
```

Comment:
this block probably doesn't need to be nested in `.pf-c-table__toggle` and just written as

```
.pf-c-table__toggle-icon {
    transition: var(--pf-c-table__toggle__icon--Transition);
    
    .pf-c-button.pf-m-expanded > & {
        transform: var(--pf-c-table__toggle--m-expanded__icon--Transform);
    }
}
```

The same could be applied to table.scss